### PR TITLE
Update docs for upgrading with terraform

### PR DESCRIPTION
**Pull request by while1eq1**
explained in commits

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/2409)

### CODE REVIEW

**Comment by while1eq1** [comment]
I signed the CLA

File: docs/upgrade.md [hunk 1]
```diff
@@ -31,3 +31,14 @@ NOTE: rolling-update does not yet perform a real rolling update - it just shuts
  there will be downtime [Issue #37](https://github.com/kubernetes/kops/issues/37)
 We have implemented a new feature that does drain and validate nodes.  This feature is experimental, and you can use the new feature by setting `export KOPS_FEATURE_FLAGS="+DrainAndValidateRollingUpdate"`.
 
+### Terraform Users
+
+* `kops edit cluster $NAME`
+* set the KubernetesVersion to the target version (e.g. `v1.3.5`)
+* `kops update cluster $NAME --target=terraform`
```
Comment (chrislovecnm) [inline docs/upgrade.md]: You can do multiple line md in github btw. Three ticks

File: cmd/kops/rollingupdatecluster.go [hunk 2]
```diff
@@ -99,6 +99,9 @@ This command updates a kubernetes cluster to match the cloud, and kops specifica
 
 To perform rolling update, you need to update the cloud resources first with "kops update cluster"
 
+Note: terraform users will need to "kops update cluster --target=terraform" then "terraform plan" then "terraform apply"
```
Comment (chrislovecnm) [inline cmd/kops/rollingupdatecluster.go]: Can we put the commands in back ticks? Your call.

File: docs/upgrade.md [hunk 3]
```diff
@@ -31,3 +31,14 @@ NOTE: rolling-update does not yet perform a real rolling update - it just shuts
...
+* `terraform plan`
```
Comment (chrislovecnm) [inline docs/upgrade.md]: Does a user have to be in a specific directory with tf?

**Comment by chrislovecnm** [comment]
@k8s-bot ok to test

**Comment by chrislovecnm** [comment]
@while1eq1 those comments with the backticks messed up Jenkins? I am asking because I will file an issue

**Comment by while1eq1** [comment]
Yeh if you look at the commits, when I added the back ticks it looks like
it broke the multi line comment in the Go code.

When I pulled them back out the build passed.

**Pull request merged**